### DISCUSSION TITLE
How to call an HTTP endpoint in Blazor Hybrid?

### DISCUSSION THREAD

**Discussion by HugoVG** [dotnet/maui#10608] [Q&A] [answered]
I've in stuck with this issue for over a couple weeks now, my situation is the following
Dotnet asp API,
Blazor Webassembly website
and a Blazor Hybrid phone app(so i could re use the website parts)

UWP version works fine on both IP and Localhost, phone however will not make ANY api calls, HTTPS or HTTP.

even adding the URL to AndroidManifest.xml and network_security_config.xml

Https gets shot down because it says that the Certificate is untrusted because it is self signed(because it is on production)

Https gets shot down for
```cs
Mixed Content: The page at 'https://0.0.0.0/' was loaded over HTTPS, but requested an insecure frame 'http://192.168.230.1:7222/'. This request has been blocked; the content must be served over HTTPS.
```
any one got help or tips?

**Comment by HugoVG** [comment, marked as answer]
after a very long battle, I came across the answer.
let me guide you through the path of insanity I came across
in your Server side make sure you have 
```cs
#if !DEBUG
app.UseHttpsRedirection();
#endif
```
you don't want to upgrade anything
in your Android Blazor Hybrid
- AndroidManifest.xml add android:usesCleartextTraffic="true"  (please be sure that you are NOT gonna use this in production code
MainPage.xaml.cs
```cs
private void Bwv_BlazorWebViewInitialized(object sender, Microsoft.AspNetCore.Components.WebView.BlazorWebViewInitializedEventArgs e)
        {
#if ANDROID && DEBUG
          e.WebView.Settings.MixedContentMode = Android.Webkit.MixedContentHandling.AlwaysAllow;
#endif
        }
```
MainPage.xaml
- <BlazorWebView HostPage="wwwroot/index.html" **BlazorWebViewInitialized="Bwv_BlazorWebViewInitialized"**>

**Comment by rajeshsws** [comment]
Hi Its working fine in Android, Do you have any solution for WIndow As well in .net maui blazor?
Thank you very much
